google/cloud/spanner/internal/partial_result_set_reader.cc

```diff
@@ -115,8 +120,36 @@ Status PartialResultSetReader::ReadFromStream() {
   }
 
   // TODO(#271) store and use resume_token.
-  // TODO(#270) handle chunked_value.
+
   values_.Swap(result_set.mutable_values());
+  // If the last response had `chunked_value` set, partial_chunked_value_
+  // contains the partial value; merge it with the first value in the
+  // response.
+  if (partial_chunked_value_.has_value()) {
+    if (values_.empty()) {
+      return Status(StatusCode::kInternal,
+                    "PartialResultSet contained no values to merge with prior "
+                    "chunked_value");
+    }
+    auto merge_status =
```

Comment:
Why do we choose to merge `values_[0]` onto `partial_chunked_value_` and then move `partial_chunked_value_` to `values_[0]`, rather than just merge `partial_chunked_value_` into `values_[0]`?  That is, should `MergeChunk()` be "onto front" rather than "onto end"?

Reply:
Maybe, what @devjgm implemented was what I told him I needed, and doing the opposite seems weird as an API (the current one is like 'operator+=' , is there an analog for the other one?  I guess it would need to be named something with "Prepend" to make sense). Under the covers we don't have much choice but to append since we're dealing with strings and vectors. Being being able to modify both arguments saves us half a copy. I'll let Greg weigh in, or we can discuss after lunch.

Reply:
We can certainly change `MergeChunk` if a different signature would be more convenient to use here, though I believe it would still need to maintain the append-like semantics.